tests/system/common.py

```diff
@@ -771,6 +778,99 @@ def running_task_status(task_statuses):
     assert False, "Did not find a TASK_RUNNING status in task statuses: %s" % (task_statuses,)
 
 
+def unreachable_tasks(framework_name='marathon'):
+    """ Returns the unreachable tasks for a given framework.
+    """
+    client = mesos.DCOSClient()
+    frameworks = client.get_master_state()['frameworks']
+    for framework in frameworks:
+        if framework["name"] == framework_name:
+            return framework['unreachable_tasks']
+    return []
+
+def mesos_tasks(framework_name='marathon'):
+    client = mesos.DCOSClient()
+    frameworks = client.get_master_state()['frameworks']
+    for framework in frameworks:
+        if framework["name"] == framework_name:
+            return framework['tasks']
+    return []
+
+
+def unreachable_task_predicate(service):
+    return len(unreachable_tasks(service)) > 0
+
+
+def elapse_time(start, end, precision=0):
+    return round(end - start, precision)
+
+
+def wait_for_unreachable_task(service='marathon', timeout_sec=10*60, inverse=False):
+
+    report_freq = 5 if timeout_sec < 600 else 15
+    print("Starting wait for unreachable tasks")
+    @retrying.retry(wait_fixed=1000, stop_max_delay=timeout_sec*1000)
+    def __wait_for_unreachable_task(start=time.time(), msg='unreachable tasks'):
+        elapsed = elapse_time(start, time.time())
+        if (elapsed % report_freq == 0):
+            print("Waiting on {} at {}s".format(msg, elapsed))
+
+        if inverse:
+            assert not unreachable_task_predicate(service)
+        else:
+            assert unreachable_task_predicate(service)
+
+    __wait_for_unreachable_task()
+
+
+def marathon_task_predicate(app_id, task_id):
+    """ Returns true if there is a task that is NOT the task_id
+    """
+    client = marathon.create_client()
+    tasks = client.get_tasks(app_id)
+    task = task_other_than_task_id(tasks, task_id)
+    return task is not None
+
+
+def task_other_than_task_id(tasks, task_id):
+    for task in tasks:
+        if task['id'] != task_id:
+            return task
+    return None
+
+
+def wait_for_marathon_task(app_id, task_id, timeout_sec=6*60):
+
+    print("Starting wait for marathon task for {} which is not task: {}".format(app_id, task_id))
+    @retrying.retry(wait_fixed=1000, stop_max_delay=timeout_sec*1000)
+    def __wait_for_marathon_task(start=time.time(), msg='marathon task'):
+        elapsed = elapse_time(start, time.time())
+        if (elapsed % 5 == 0):
+            print("Waiting on {} at {}s".format(msg, elapsed))
+
+        assert marathon_task_predicate(app_id, task_id)
+
+    __wait_for_marathon_task()
+
+
+def task_remove_predicate(app_id, task_id):
```

Comment:
I'm not sure that this is tested in the method. It seems to test that there is only one task. Not that the task has been removed.